### DISCUSSION TITLE
ITead Sonoff Zigbee 3.0 USB Dongle Plus will be based on TI CC2652P

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#816] [General]
FYI, ITead has just announced on Facebook that their  CC2652P based "Sonoff Zigbee 3.0 USB Dongle Plus" will be released soon.

https://sonoff.tech/new-product-coming-soon/?fbclid=IwAR2nyXNbxQ2K0-yIx7ccPtG6OAY_6hJkVG6xHwdnRfU23AYvfXFREDagCjQ

According to their marketing material it will come pre-flashed with Texas Instruments Z-Stack 3.x.0 coordinator firmware and it should work out-of-the-box with either Home Assistant's ZHA (zigpy) and Zigbee2MQTT (zigbee-herdsman, so probably Jeedom and IoBroker too then.

In addition, they specifically mention it can also function as a Zigbee router (presumably by flashing [other firmware](https://github.com/Koenkk/Z-Stack-firmware/) and open access buttons).

I also read that this time they have also added a proper electromagnetic shielding to the radio chip and antenna parts onboard the board itself, meaning they must have learned from some of their design flaws in regards to electromagnetic interference and radio signal reception.

https://www.facebook.com/SONOFF.official/posts/2867330886911891

https://www.facebook.com/iteadstudio/posts/4960944183932977

It is based on Texas Instruments CC2652P (CC2652 with integrated +20 dBm amplifier) Zigbee radio and features a metal casing + an SMA connector with an external antenna. It looks a little on the large side for USB 2.0 Type-A but still recommend using a USB extension cable.

![image](https://user-images.githubusercontent.com/6320001/134296915-7131efa5-8319-4d5d-a9a7-dd85116caafb.png)

![image](https://user-images.githubusercontent.com/6320001/134298087-d55c19fc-89a4-479c-b4e9-9950ffd49f12.png)

![image](https://user-images.githubusercontent.com/6320001/134298156-07c4500b-33f5-482a-b5fc-be4f0a1de0c7.png)

Interestingly it uses a Silabs CP2102N UART-to-USB chip so wonder if it will have ITead VID and/or PID strings for auto USB discovery -> https://github.com/zigpy/zigpy/discussions/808

Also, wondering if this CC2652P serial port will have pins connected for RTS / CTS Hardware Flow Control?

**Comment by MattWestb** [comment]
On the boarder to over kill with metal casing but looks nice !!
If they have doing one simple metal RF-child on the MG21 for some cents it have being one great product that have working very well but was one big flop in the end after the problems our advance ZHA user was having (no named no blamed user).

**Comment by Hedda** [comment]
By the way, I understand ITead went with CC2652P instead of EFR32MG21 for this "Plus" version because of the current silicon chip shortage.

**Reply by MattWestb** [reply]
IKEA, X-Philips and tuy is some of the vacuum cleaner off the MG21 chips / modules on the market.
I was writing it in the IKEA 3 gen that IKEA have some FCC applications that is over 2.5 year and not have putting them in production and very likely is being shortages of the Silabs module (or the price Silabs was offering them).

So its not only Scania trucks, Volvo Cars and trucks that have chip shortages for getting there products on the market ;-)

By the way Gary is also waiting for getting some MG21 modules for very long time but i hope the Christmas man is bringing it to him !!

**Reply by Hedda** [reply]
Home Assistant Amber developers also mentioned in their introduction and Q&A vidfeo that the Home Assistant Amber release date is only set so far into the future specifically because of Silabs MGM210P module worldwide component shortage for silicon chips

https://www.home-assistant.io/blog/2021/09/13/home-assistant-amber/

https://www.youtube.com/watch?v=KPR-shzT_VQ

"**_Orders placed now ship Jun 14, 2022._**"

https://www.crowdsupply.com/nabu-casa/home-assistant-amber

**Comment by Hedda** [comment]
> Also, wondering if this CC2652P serial port will have pins connected for RTS / CTS Hardware Flow Control?

They now posted some documentation which states that the hardware does support Hardware Flow Control but default firmware does not:

https://sonoff.tech/product-review/zigbee-3-0-usb-dongle-plus/

"_The dongle is pre-flashed with the official Zigbee 3.0 coordinator firmware, which does not support software flow control. The dongle supports hardware flow control, if you want to enable it, please set the dip switch to on, and generate the firmware that supports hardware flow control before running, see the following document for details on how to generate the corresponding firmware._"

https://sonoff.tech/wp-content/uploads/2021/09/Zigbee-3.0-USB-dongle-plus-firmware-flashing-.docx

_**Enable hardware flow control and generate firmware (optional)** If you need to enable the hardware flow control of the CC2652P USB Dongle, you need to use CCS to import the ZNP project to configure and compile the firmware that supports the hardware flow control._

**Comment by Hedda** [comment]
FYI, the initial batch was released but already sold out, however pre-orders for next batch is now available at Itead’s official store.

https://itead.cc/product/sonoff-zigbee-3-0-usb-dongle-plus/

https://sonoff.tech/product/diy-smart-switch/sonoff-dongle-plus/

![image](https://user-images.githubusercontent.com/6320001/135108630-e8d8f76c-b0df-4bf2-a5ec-ee72b103e525.png)

![image](https://user-images.githubusercontent.com/6320001/135108645-3db38718-3770-4342-a467-8fdc22a97e6f.png)

![image](https://user-images.githubusercontent.com/6320001/135108716-7cdffc6c-f419-4d0d-8821-e80706fdc138.png)

![image](https://user-images.githubusercontent.com/6320001/135108748-529e7276-4ce7-43e1-af4b-db36a9b4b208.png)

**Comment by Hedda** [comment]
Koenkk from Zigbee2MQTT got info from ITead/Sonoff that this dongle comes pre-flashed with older "CC1352P2_CC2652P_launchpad_coordinator_20210120" firmware from https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin so specifications listed in his readme file should still apply -> https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/README.md

**Comment by Hedda** [comment]
FYI, Koenkk has now uploaded an experimental firmware for CC2652P/CC1352P with a default set to 9 dBm (and a max of 20 dBm), see/download: [https://github.com/Koenkk/Z-Stack-firmware/issues/323](https://github.com/Koenkk/Z-Stack-firmware/issues/323)

He mentions in [https://github.com/Koenkk/zigbee2mqtt/issues/8885](https://github.com/Koenkk/zigbee2mqtt/issues/8885) that he does not himself have RF measurement equipment so cannot confirm RF output power is exactly as per config set in software.
